### ISSUE TITLE
Exception encountered when building app with a non-JS import handled by plugin loader

### ISSUE THREAD

**Issue by dkrill** [systemjs/builder#498]
After updating from systemjs-builder#0.14.11 to any newer version (0.14.12 <= 0.15.7), we encountered an exception thrown by `builder.buildStatic`. 

```
[ERROR] Bundle failed: 
 ›› MultipleErrors: template.sidebar.dust:1:1: Unexpected token <
template.sidebar.dust:1:2: Semi-colon expected
template.sidebar.dust:1:6: Unexpected token <
template.sidebar.dust:1:7: Semi-colon expected
template.sidebar.dust:1:16: Semi-colon expected
template.sidebar.dust:1:29: Unexpected token <
...
```

`template.sidebar.dust` is a dust.js resource being loaded via a custom plugin. 

```
import template from './template.sidebar.dust!boneplate';

export default function() {
  return template;
}
```

The plugin compiles and registers our template with dust, then outputs a module that exports the name of the template that it was registered with dust.  

```
import compiler from 'dustjs-compiler';
import dust from 'dustjs-linkedin';

function getTemplateName(name) {
  // return name without the file extension
}

export function translate(load) {
  const templateName = getTemplateName(load.address.substr(load.address.lastIndexOf('/') + 1));

  const template = load.source;
  let compiled;
  if (template.indexOf('(function(){dust.register') === 0 || template.indexOf('(function(dust){dust.register') === 0) {
    compiled = template;
  } else {
    compiled = compiler.compile(template, templateName);
  }
  dust.loadSource(compiled);

  return 'import dust from "dustjs-linkedin";' + compiled + 'export default "' + templateName + '";';
}
```

In 0.14.11, the build succeeds.  In 0.14.12 and later, something appears to be parsing the dust file source as script.

I've created an example app that duplicates the issue:
https://github.com/dkrill/systemjs-builder-plugin-issue

Run `npm install`, then `gulp build`.  Decrement the version of `systemjs-builder` to 0.14.11 in `package.json` and the exception goes away.  The version of `systemjs` used doesn't appear to be a factor.

**Comment by guybedford** [comment]
Thanks for sharing the replication here. I had a look at the test case but it seems to rely on a Bitbucket registry in order to install correctly?

The 0.14.12 release notes are at https://github.com/systemjs/builder/releases/tag/0.14.12, I can't immediately see anything there that would have affected this but it does look like there were changes to the plugin system.

If you can get the replication working without requiring Bitbucket (or at least any sort of private access if that is the case?), I'll gladly look into this further.

**Comment by dkrill** [comment]
I've removed the errant bitbucket reference. Sorry about that.  

I just wanted to point out that the exact error I'm getting in this simplified test case is different than the one we see in our original project.  

```
[11:44:37] 'build' errored after 1.05 s
[11:44:37] TypeError: invalid data
    at WriteStream.Socket.write (net.js:617:11)
    at ~/.../systemjs-builder-plugin-issue/gulpfile.babel.js:13:22
```

However, the root cause appears to be the same.  Either downgrading to 0.14.11 or removing the plugin-loaded resource import eliminates the error.  I'm not sure if it means my test isn't representative of the true problem, or if it's a side-effect of the paring down I did.  If you find it to be the former, let me know and I'll make the needed adjustments.

I appreciate your attention on this.

**Comment by guybedford** [comment]
Here's a hotfix - 

SystemJS-Builder 0.14.15, in `lib/trace.js` line 368, replace the conditional:

```javascript
if (load.metadata.format == 'esm')
  load.source = originalSource;
```

```javascript
if (load.metadata.format == 'esm' && !load.metadata.loaderModule)
  load.source = originalSource;
```

The 0.14 branch is no longer being maintained, but that should at least allow getting to the latest there.

**Comment by dkrill** [comment]
Confirming that your hotfix resolved the issue.  

Will this same fix need to be applied to the 0.15 branch?  If so, we'll upgrade from 0.14 to the latest stable after the next patch release.

**Comment by guybedford** [comment]
I checked and the bug doesn't seem to be on the 0.15 branch as far as I can tell. Yes it would be advisable to upgrade when you can.
